### DISCUSSION TITLE
Web component in next.js

### DISCUSSION THREAD

**Discussion by truefalse10** [vercel/next.js#15570] [Help]
Hi,
i am trying to use a web component in my next.js project. My page looks something like this:
```
export const Home = () => (
<div>
    <webcom-input-field label="Required"></webcom-input-field>
    <div
      dangerouslySetInnerHTML={{
        __html: "<webcom-input-field label="Required (dangerous)"></webcom-input-field>",
      }}
    ></div>
    <script src="/libs/webcom-input-field.js"></script>
</div>)
```
here is what happens on the client:
1. the first element does not get rendered and i get a client-side error with `react-dom.development.js:88 Warning: Did not expect server HTML to contain a <span> in <webcom-input-field>.
    in webcom-input-field (at pages/index.tsx:41)`
1. the second element (dangerouslySetInnerHTML) gets rendered correctly but i get the warning `Warning: Prop `dangerouslySetInnerHTML` did not match. Server: "\n    <webcom-input-field la...` because the tree differs on server and client

Is there an official way of integrating web components into next.js? or maybe a solution for preventing react to verify the tree of custom elements?

**Comment by ianldgs** [comment]
Although probably not the most elegant solution, you could maybe try `suppressHydrationWarning={true}`.
WebComponents are not known for working on server-side, especially together with React, for many reasons.

**Reply by igorlino** [reply]
If it may help anyone, a bit more detailed explanation how it may work:  https://css-tricks.com/using-web-components-with-next-or-any-ssr-framework/
